examples/levels/src/reverse.ts

```diff
@@ -0,0 +1,237 @@
+
+interface BondingLevel {
+    price: number;
+    quantity: number;
+    level: number;
+}
+
+/**
+ * Generates order levels similar to a bonding curve from AMMs 
+ * Returns the bonding levels corresponding prices and quantities.
+ * Price increases and quantity decreases are reduced at each level.
+ */
+function generateBondingLevels(
+    totalSupply: number = 1_000_000_000,
+    numLevels: number = 100,
+): BondingLevel[] {
+    const levels: BondingLevel[] = [];
+    const initialPrice = INITIAL_FDV / totalSupply;
+    
+    let currentQuantity = totalSupply * INITIAL_QTY_RATIO;
+    let currentPrice = initialPrice;
+    
+    for (let level = 0; level < numLevels; level++) {
+        levels.push({
+            price: currentPrice,
+            quantity: currentQuantity,
+            level: level + 1,
+        });
+        const currentPriceIncrease = Math.max(INITIAL_PRICE_INCR - (level * PRICE_INCR_REDUCTION), 0);
+        const currentQuantityReduction = Math.min(0, INITIAL_QTY_REDUCTION - (level * QTY_DECR_REDUCTION));
+        
+        currentQuantity *= (1 + currentQuantityReduction);
+        currentPrice *= (1 + currentPriceIncrease);
+    }
+
+    console.log('Price Range:', {
+        initialPrice,
+        finalPrice: levels[levels.length - 1].price,
+        priceMultiple: levels[levels.length - 1].price / initialPrice,
+        numLevels: levels.length
+    });
+    
+    return levels;
+}
+
+/**
+ * Gets the current level based on total tokens sold.
+ * Levels are 1-based to match the order ID convention.
+ */
+function getCurrentLevel(
+    totalTokensSold: number,
+    levels: BondingLevel[]
+): number {
+    let accumulatedTokens = 0;
+    
+    for (let i = 0; i < levels.length; i++) {
+        accumulatedTokens += levels[i].quantity;
+        if (accumulatedTokens >= totalTokensSold) {
+            return i + 1;
+        }
+    }
+    
+    return levels.length;
+}
+
+/**
+ * Generates initial orders for the bonding curve using reverse orders.
+ * Each order will automatically flip between bid/ask after being filled.
+ */
+function generateInitialOrders(
+    levels: BondingLevel[],
+    currentLevel: number
+): WrapperPlaceOrderParamsExternal[] {
+    const orders: WrapperPlaceOrderParamsExternal[] = [];
+    
+    // Place bid orders for cleared levels
+    for (let i = 1; i < currentLevel; i++) {
+        const level = levels[i - 1];
+        orders.push({
+            numBaseTokens: level.quantity,
+            tokenPrice: level.price,
+            isBid: true,
+            lastValidSlot: 0,
+            orderType: OrderType.Reverse,  // Use reverse orders
+            clientOrderId: i
+        });
+    }
+
+    // Place ask orders for remaining levels
+    for (let i = currentLevel; i <= levels.length; i++) {
+        const level = levels[i - 1];
+        orders.push({
+            numBaseTokens: level.quantity,
+            tokenPrice: level.price,
+            isBid: false,
+            lastValidSlot: 0,
```

Comment:
No. Use the new version that has a spread.

Reply:
I can make the sdk block you from doing this if it was confusing which way it was supposed to be done

Reply:
good catch forgot to test this